go-runtime/ftl/reflection/type_registry.go

```diff
@@ -66,10 +50,21 @@ func NewTypeRegistry(options ...TypeRegistryOption) *TypeRegistry {
 	return t
 }
 
+// Register applies all the provided options to the singleton TypeRegistry
+func Register(options ...TypeRegistryOption) {
```

Comment:
Now that this is a global function, we could probably get rid of the `TypeRegistryOption` concept and just have:

```go
func RegisterSumType[Discriminant any](variants...Discriminant)
```

Current approach is totally fine though, I'm just thinking out loud :)

Reply:
Nice! Done :D